tiledb/sm/array_schema/dimension.cc

```diff
@@ -1679,3 +1646,45 @@ Range DimensionVarSize::compute_mbr_var(
 }
 
 }  // namespace tiledb::sm
+
+std::string tile_extent_str(const tiledb::sm::Dimension& dim) {
+  std::stringstream ss;
+
+  if (!dim.tile_extent()) {
+    return tiledb::sm::constants::null_str;
+  }
+
+  auto g = [&](auto T) {
+    if constexpr (tiledb::type::TileDBNumeric<decltype(T)>) {
+      auto val = reinterpret_cast<const decltype(T)*>(dim.tile_extent().data());
+      ss << *val;
+      return ss.str();
+    }
+    throw std::logic_error(
+        "Datatype::" + datatype_str(dim.type()) + " not supported");
```

Comment:
It was arguably a mistake to `throw` here. The problem is that if there is somehow an invalid or unsupported type here, it should output a message that says so, not generate an error that only appears in the log.